src/zc/relation/tokens.txt

```diff
@@ -60,8 +60,97 @@ Now we can create some classes.  In the README example, the setup was a bit
 of a toy.  This time we will be just a bit more practical.  We'll also expect
 to be operating within the ZODB, with a root and transactions. [#ZODB]_
 
+.. [#ZODB] Here we will set up a ZODB instance for us to use.
+
+    >>> from ZODB.tests.util import DB
+    >>> db = DB()
+    >>> conn = db.open()
+    >>> root = conn.root()
+
 Here's how we will dump and load our relations: use a "registry"
-object, similar to an intid utility. [#faux_intid]_  
+object, similar to an intid utility. [#faux_intid]_
+
+.. [#faux_intid] Here's a simple persistent keyreference.  Notice that it is
+    not persistent itself: this is important for conflict resolution to be
+    able to work (which we don't show here, but we're trying to lean more
+    towards real usage for this example).
+
+    >>> class Reference(object): # see zope.app.keyreference
+    ...     def __init__(self, obj):
+    ...         self.object = obj
+    ...     def _get_sorting_key(self):
+    ...         # this doesn't work during conflict resolution.  See
+    ...         # zope.app.keyreference.persistent, 3.5 release, for current
+    ...         # best practice.
+    ...         if self.object._p_jar is None:
+    ...             raise ValueError(
+    ...                 'can only compare when both objects have connections')
+    ...         return self.object._p_oid or ''
+    ...     def __lt__(self, other):
+    ...         # this doesn't work during conflict resolution.  See
+    ...         # zope.app.keyreference.persistent, 3.5 release, for current
+    ...         # best practice.
+    ...         if not isinstance(other, Reference):
+    ...             raise ValueError('can only compare with Reference objects')
+    ...         return self._get_sorting_key() < other._get_sorting_key()
+    ...     def __eq__(self, other):
+    ...         # this doesn't work during conflict resolution.  See
+    ...         # zope.app.keyreference.persistent, 3.5 release, for current
+    ...         # best practice.
+    ...         if not isinstance(other, Reference):
+    ...             raise ValueError('can only compare with Reference objects')
+    ...         return self._get_sorting_key() == other._get_sorting_key()
```

Comment:
I don't like seeing only two of six rich comparison methods being defined. Can you add the @functools.total_ordering class decorator?